### PR TITLE
Fix genesis mainnet issues

### PR DESCRIPTION
**Pull request by felipemadero**
Closes https://github.com/ava-labs/avalanche-cli/issues/1197 https://github.com/ava-labs/avalanche-cli/issues/1200 https://github.com/ava-labs/avalanche-cli/issues/1202 https://github.com/ava-labs/avalanche-cli/issues/1201 https://github.com/ava-labs/avalanche-cli/issues/1235 https://github.com/ava-labs/avalanche-cli/issues/1234

### CODE REVIEW

File: cmd/subnetcmd/deploy.go [hunk 1]
```diff
@@ -187,62 +184,82 @@ func runDeploy(cmd *cobra.Command, args []string) error {
...
 		return err
 	}
-	return app.WriteGenesisMainnetFile(chain, prettyJSON.Bytes())
-}
-
-func handleMainnetChainID(chain string) error {
-	genesisMainnetPath := app.GetGenesisMainnetPath(chain)
-	_, err := os.ReadFile(genesisMainnetPath)
-	if err != nil && os.IsNotExist(err) {
+	if evmGenesis.Config == nil {
```
Comment (arturrez) [inline cmd/subnetcmd/deploy.go:212]: nit: can be combined into `if evmGenesis.Config == nil || evmGenesis.Config.ChainID == nil`

File: cmd/subnetcmd/deploy.go [hunk 2]
```diff
@@ -187,62 +184,82 @@ func runDeploy(cmd *cobra.Command, args []string) error {
...
 		newChainIDPrompt := "Using the same ChainID for both Fuji and Mainnet could lead to a replay attack. Do you want to use a different ChainID?"
-		var decision string
-		if mainnetChainID == "" && os.Getenv(constants.SimulatePublicNetwork) == "" {
-			decision, err = app.Prompt.CaptureList(newChainIDPrompt, listOptions)
-			if err != nil {
-				return err
-			}
+		var (
+			err      error
+			decision string
```
Comment (arturrez) [inline cmd/subnetcmd/deploy.go:231]: don't need this just use `decision, err := app.Prompt.CaptureList(newChainIDPrompt, listOptions)`

File: cmd/subnetcmd/deploy.go [hunk 3]
```diff
@@ -187,62 +184,82 @@ func runDeploy(cmd *cobra.Command, args []string) error {
...
 	}
-	var prettyJSON bytes.Buffer
-	err = json.Indent(&prettyJSON, jsonBytes, "", "    ")
+	config["chainId"] = float64(newChainID)
+	return json.MarshalIndent(genesisMap, "", "  ")
+}
+
+// updates sidecar with genesis mainnet id to use
+// given either by cmdline flag, original genesis id, or id obtained from the user
+func getSubnetEVMMainnetChainID(sc *models.Sidecar, subnetName string) error {
```
Comment (arturrez) [inline cmd/subnetcmd/deploy.go:206]: nit: rename sc to sidecar

File: cmd/subnetcmd/deploy.go (hunk 1 above)
Comment (felipemadero) [reply cmd/subnetcmd/deploy.go:212]: It should be &&. But I prefer to keep separate error msgs.

File: cmd/subnetcmd/export.go [hunk 4]
```diff
@@ -162,7 +126,7 @@ func exportSubnet(_ *cobra.Command, args []string) error {
 		}
 	}
 
-	gen, err := app.LoadRawGenesis(subnetName, network)
+	gen, err := app.LoadRawGenesis(subnetName)
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply cmd/subnetcmd/export.go:129]: It is just going to use the original genesis, but is also exporting the sidecar, so the mainnet chain id is kept.

**Pull request merged**